packages/ra-core/src/types.ts

```diff
@@ -361,6 +361,10 @@ export interface ResourceProps {
     children?: ReactNode;
 }
 
+export type RenderRecordFunction<
+    RecordType extends Record<string, unknown> = any
+> = (record: RecordType, source?: string) => ReactNode;
+
```

Comment:
Let me know if there’s a more appropriate place to put this. Named to match the existing `RenderResourcesFunction`:

https://github.com/marmelab/react-admin/blob/b007f35f98a4046a146cd51799f2829b9020cb89/packages/ra-core/src/types.ts#L303